tedana/tests/test_integrity _robustica.py

```diff
@@ -0,0 +1,667 @@
+
+    if sorted(found_files) != sorted(expected_files):
+        expected_not_found = sorted(list(set(expected_files) - set(found_files)))
+        found_not_expected = sorted(list(set(found_files) - set(expected_files)))
+
+        msg = ""
+        if expected_not_found:
+            msg += "\nExpected but not found:\n\t"
+            msg += "\n\t".join(expected_not_found)
+
+        if found_not_expected:
+            msg += "\nFound but not expected:\n\t"
+            msg += "\n\t".join(found_not_expected)
+        raise ValueError(msg)
+
+
+def data_for_testing_info(test_dataset=str):
+    """
+    Get the path and download link for each dataset used for testing.
+
+    Also creates the base directories into which the data and output
+    directories are written
+
+    Parameters
+    ----------
+    test_dataset : str
+       References one of the datasets to download. It can be:
+        three-echo
+        three-echo-reclassify
+        four-echo
+        five-echo
+
+    Returns
+    -------
+    test_data_path : str
+       The path to the local directory where the data will be downloaded
+    osf_id : str
+       The ID for the OSF file.
+       Data download link would be https://osf.io/osf_id/download
+       Metadata download link would be https://osf.io/osf_id/metadata/?format=datacite-json
+    """
+
+    tedana_path = os.path.dirname(tedana_cli.__file__)
+    base_data_path = os.path.abspath(os.path.join(tedana_path, "../../.testing_data_cache"))
+    os.makedirs(base_data_path, exist_ok=True)
+    os.makedirs(os.path.join(base_data_path, "outputs"), exist_ok=True)
+    if test_dataset == "three-echo":
+        test_data_path = os.path.join(base_data_path, "three-echo/TED.three-echo")
+        osf_id = "rqhfc"
+        os.makedirs(os.path.join(base_data_path, "three-echo"), exist_ok=True)
+        os.makedirs(os.path.join(base_data_path, "outputs/three-echo"), exist_ok=True)
+    elif test_dataset == "three-echo-reclassify":
+        test_data_path = os.path.join(base_data_path, "reclassify")
+        osf_id = "f6g45"
+        os.makedirs(os.path.join(base_data_path, "outputs/reclassify"), exist_ok=True)
+    elif test_dataset == "four-echo":
+        test_data_path = os.path.join(base_data_path, "four-echo/TED.four-echo")
+        osf_id = "gnj73"
+        os.makedirs(os.path.join(base_data_path, "four-echo"), exist_ok=True)
+        os.makedirs(os.path.join(base_data_path, "outputs/four-echo"), exist_ok=True)
+    elif test_dataset == "five-echo":
+        test_data_path = os.path.join(base_data_path, "five-echo/TED.five-echo")
+        osf_id = "9c42e"
+        os.makedirs(os.path.join(base_data_path, "five-echo"), exist_ok=True)
+        os.makedirs(os.path.join(base_data_path, "outputs/five-echo"), exist_ok=True)
+    else:
+        raise ValueError(f"{test_dataset} is not a valid dataset string for data_for_testing_info")
+
+    return test_data_path, osf_id
+
+
+def download_test_data(osf_id, test_data_path):
```

Comment:
Are these data identical to those used in existing TEDANA tests for evaluating with use of FastICA? If so, it would be preferable if executing *either* FastICA or RobustICA tests would download the test data, but executing *both* tests would download those data *only once*.

Reply:
Addition: Even if this turns out to be the current behaviour based on the subsequent code, it is also the case that the code responsible for doing so should appear only once, rather than being duplicated across the FastICA and RobustICA tests.